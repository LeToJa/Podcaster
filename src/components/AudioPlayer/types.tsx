export interface AudioPlayerTypes {
    url: string
    type: string
}