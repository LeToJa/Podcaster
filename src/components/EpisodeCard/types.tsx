export interface EpisodeCardTypes {
    podcastId: string
    episodeId: number
    title: string
    date: string
    duration: string
}