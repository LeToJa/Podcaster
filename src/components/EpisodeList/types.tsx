import { PodcastEpisode } from '../../layout/PodcastLayout/types'

export interface EpisodeListTypes {
    podcastId: string
    episodes: PodcastEpisode[]
}