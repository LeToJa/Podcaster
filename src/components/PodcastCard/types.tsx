export interface PocastCardTypes {
    id: string
    name: string
}