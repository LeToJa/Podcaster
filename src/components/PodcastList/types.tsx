import { PocastCardTypes } from '../PodcastCard/types'

export interface PodcastListTypes {
    podcasts: PocastCardTypes[]
}