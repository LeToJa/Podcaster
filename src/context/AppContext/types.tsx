export interface AppContextTypes {
    loading: boolean
    toggleLoading: (value: boolean) => void
}