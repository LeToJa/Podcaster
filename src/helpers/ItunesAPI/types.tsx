export interface FeedResponseTypes {
    feed: {
        entry: {
            id: {
                attributes: {
                    'im:id': string
                }
            }
            'im:name': { label: string }
            'im:artist': { label: string }
            'im:image': { label: string }[]
        }[]
    }
}

export interface DefaultResponseFetch {
    contents: string
}

export interface PodcastResponseTypes {
    resultCount: number;
    results: (PodcastResultsEntity)[];
}
interface PodcastResultsEntity {
    wrapperType: string;
    kind: string;
    artistId: number;
    collectionId: number;
    trackId: number;
    artistName: string;
    collectionName: string;
    trackName: string;
    collectionCensoredName: string;
    trackCensoredName: string;
    artistViewUrl: string;
    collectionViewUrl: string;
    feedUrl: string;
    trackViewUrl: string;
    artworkUrl30: string;
    artworkUrl60: string;
    artworkUrl100: string;
    collectionPrice: number;
    trackPrice: number;
    collectionHdPrice: number;
    releaseDate: string;
    collectionExplicitness: string;
    trackExplicitness: string;
    trackCount: number;
    trackTimeMillis: number;
    country: string;
    currency: string;
    primaryGenreName: string;
    contentAdvisoryRating: string;
    artworkUrl600: string;
    genreIds?: (string)[] | null;
    genres?: (string)[] | null;
}
