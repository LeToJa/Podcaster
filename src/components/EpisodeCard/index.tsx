import { Link } from 'react-router-dom'

import { EpisodeCardTypes } from './types'

const EpisodeCard = ({ podcastId, episodeId, title, date, duration }: EpisodeCardTypes) => <>
    <tr data-testid='podcast-episode'>
        <td colSpan={8}>
            <Link to={`/podcast/${podcastId}/episode/${episodeId}`} className='font-bold text-sky-600'>
                {title}
            </Link>
        </td>
        <td className='text-center'>{date}</td>
        <td className='text-center'>{duration}</td>
    </tr>
</>

export default EpisodeCard