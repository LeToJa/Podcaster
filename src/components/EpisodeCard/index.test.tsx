import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { BrowserRouter } from 'react-router-dom'

import { EpisodeCardTypes } from './types'

import EpisodeCard from '.'

const mockEpisode: EpisodeCardTypes = {
    podcastId: '123',
    episodeId: 456,
    title: 'Sample Episode',
    date: '2024-08-30',
    duration: '45:30'
}

describe('EpisodeCard', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <table>
                    <tbody>
                        <EpisodeCard {...mockEpisode} />
                    </tbody>
                </table>
            </BrowserRouter>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders with correct content and link', () => {
        const { getByText } = render(
            <BrowserRouter>
                <table>
                    <tbody>
                        <EpisodeCard {...mockEpisode} />
                    </tbody>
                </table>
            </BrowserRouter>
        )

        const titleLink = getByText('Sample Episode')
        expect(titleLink).toBeTruthy()
        expect(titleLink.closest('a')).toHaveProperty('href', 'http://localhost:3000/podcast/123/episode/456')

        expect(getByText('2024-08-30')).toBeTruthy()
        expect(getByText('45:30')).toBeTruthy()
    })
})
