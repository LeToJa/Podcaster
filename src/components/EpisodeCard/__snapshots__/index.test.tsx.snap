// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EpisodeCard > renders correctly and matches snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr
        data-testid="podcast-episode"
      >
        <td
          colspan="8"
        >
          <a
            class="font-bold text-sky-600"
            href="/podcast/123/episode/456"
          >
            Sample Episode
          </a>
        </td>
        <td
          class="text-center"
        >
          2024-08-30
        </td>
        <td
          class="text-center"
        >
          45:30
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
